plugins/modules/device.py

```diff
@@ -97,6 +97,12 @@
     required: false
     type: str
     version_added: "3.0.0"
+  software_version:
+    description:
+      - The software version of the device
+    required: false
+    type: str
```

Comment:
```suggestion
    type: raw
```

You have `type="raw"` in the `argument_spec` below, which I believe is correct. Using `raw` allows you to send in just a version as a string or key/value pairs when the same version string may be used across multiple platforms:

```yaml
            software_version:
              version: "3.2.1"
              platform: "Cisco IOS"
```